src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -2060,7 +2060,7 @@ public <T extends JCTree> T translate(T tree) {
         } else {
             make_at(tree.pos());
             T result = super.translate(tree);
-            if (endPosTable != null && result != tree) {
+            if (endPosTable != null && result != null && result != tree) {
```

Comment:
I wonder how this extra check for `result != null` was added. I mean, of course calling `replaceTree` with a null parameter seems suspicious -- should we add more formal null checks in the various end pos table methods?

Reply:
I see -- so this started to happen now because before we used to just do nothing if the empty end pos table was set. Makes sense.